macro-mathjax-ui/src/main/resources/Macros/MathJaxMacro.xml

```diff
@@ -187,6 +174,49 @@ y = \frac{c}{cb-ad}
       load: ['[tex]/noerrors']
     }
   };
+
+  /**
+   * Reset the tex labels, since re-typesetting might create duplicate labels, and perform an asynchronous typesetting.
+   * See https://docs.mathjax.org/en/latest/web/typeset.html#handling-asynchronous-typesetting.
+   */
+  var clearAndTypesetPromise = function(elems) {
+    let typesetPromise = Promise.resolve();
+    MathJax.texReset();
+    if (elems != undefined) {
+      // This could lead to bugs https://github.com/mathjax/MathJax/issues/2999, so for now typesetting specific
+      // elements should be used only if equations numbering does not count. For example, this is used for typesetting
+      // in rendered diff view, since only one math element is send at a time.
+      typesetPromise = MathJax.typesetPromise(elems);
+    } else {
+      typesetPromise = MathJax.typesetPromise();
```

Comment:
Aren't these two line equivalent when ``elems`` is undefined? In other words, is the IF really needed?